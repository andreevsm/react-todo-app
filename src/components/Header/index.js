import React from 'react';
import PropTypes from 'prop-types';

const Header = ({
  toDo,
  done,
}) => (
  <h1>
    My todo list
    {toDo}
    {' '}
    {done}
  </h1>
);

Header.propTypes = {
  toDo: PropTypes.number.isRequired,
  done: PropTypes.number.isRequired,
};

export default Header;
