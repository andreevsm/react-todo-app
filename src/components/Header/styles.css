.header {
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
  align-items: flex-end;
  margin-bottom: 1rem;
}

.header__todo-info {
  font-weight: 600;
  font-size: 1.2rem;
}

.header__title {
  margin: 0;
}

.todo-info__count {
  color: #ccc;
}
