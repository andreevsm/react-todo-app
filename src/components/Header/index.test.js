import React from 'react';
import renderer from 'react-test-renderer';

import Header from './index';

test('Header should be shown', () => {
  const component = renderer.create(
    <Header toDo={3} done={2} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
