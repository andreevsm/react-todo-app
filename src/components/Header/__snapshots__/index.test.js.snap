// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should be shown 1`] = `
<div
  className="header"
>
  <h1
    className="header__title"
  >
    My Todo list
  </h1>
  <div
    className="header__todo-info"
  >
    <div
      className="todo-info__count"
    >
      To do
      <span
        className="count"
      >
        3
      </span>
    </div>
    <div
      className="todo-info__count"
    >
      Done
      <span
        className="count"
      >
        2
      </span>
    </div>
  </div>
</div>
`;
